src/v/security/audit/schemas/schemas.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "types.h"
+
+namespace security::audit {
+
+template<typename T>
+ss::sstring rjson_serialize(const T& v) {
+    ::json::StringBuffer str_buf;
+    ::json::Writer<::json::StringBuffer> wrt(str_buf);
+
+    using ::json::rjson_serialize;
+    using ::security::audit::rjson_serialize;
+
+    rjson_serialize(wrt, v);
+
+    return ss::sstring{str_buf.GetString(), str_buf.GetSize()};
+}
+
+class ocsf_base_event {
+public:
+    auto equality_fields() const {
```

Comment:
This isn't virtual or private, so you'll get different results depending on whether you use:
* `event->hash()`
* `std::hash(base&)`
* `std::hash(derived&)`

And that will be a fun debugging session.

Reply:
🎵 engineers just want to have fun 🎵  (sang to the tune of girls just want to have fun)

Reply:
I can't make `equality_fields` virtual as the return types are different.  I think I'll move it to `protected` and change the name.